lldb/source/Expression/DWARFExpression.cpp

```diff
@@ -1106,6 +1124,25 @@ llvm::Expected<Value> DWARFExpression::Evaluate(
         return llvm::createStringError(
             "Invalid address size for DW_OP_deref_size: %d\n", size);
       }
+
+      // Deref a register or implicit location and truncate the value to `size`
+      // bytes. See the corresponding comment in DW_OP_deref for more details on
+      // why we deref these locations this way.
+      if (dwarf4_location_description_kind == Register ||
+          dwarf4_location_description_kind == Implicit) {
+        // Reset context to default values.
+        dwarf4_location_description_kind = Memory;
+        stack.back().ClearContext();
+
+        // Truncate the value on top of the stack to *size* bytes then
+        // extend to the size of an address (e.g. generic type).
+        Scalar scalar = stack.back().GetScalar();
+        scalar.TruncOrExtendTo(size * 8, /*sign=*/false);
+        scalar.TruncOrExtendTo(opcodes.GetAddressByteSize() * 8,
+                               /*sign=*/false);
+        stack.back().GetScalar() = scalar;
+        break;
+      }
```

Comment:
We should reuse the `Evaluate_DW_OP_deref` function call from line 864 and pass in the the deref size as a new argument. So pass in `size` to `Evaluate_DW_OP_deref` for this `DW_OP_deref_size`. For the actual `DW_OP_deref` we can pass in the register size.

Reply:
@clayborg Yeah, Its not nice that we have two nearly identical implementations of deref. There are some differences between them, but they may not be important. For example, DW_OP_deref reads the FileAddress from the process after converting to a load address, but DW_OP_deref_size reads a FileAddress from the target.

Let me create a separate PR to unify the implementations.